kops 1.11 beta1 SIGSEGV

**1. What `kops` version are you running?** Version 1.11.0-beta.1 (git-0c5a0a57d)

**2. What Kubernetes version are you running?** 1.11.5

**3. What cloud provider are you using?** AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

```bash
$ kops replace \
    --filename=/tmp/kluster-deploy/templates/kops/rendered/cluster.cspdev-red-us-west-2.cspdev.something.com.yaml \
    --name=cspdev-red-us-west-2.cspdev.something.com \
    --state=s3://st-cloud-cspdev \
    --v=2
```

I'd have to greek the cluster.yaml file to share it out, and I'm not sure what would be required to work.  I could share it privately if needed.

**5. What happened after the commands executed?**

```
I1129 21:01:36.579689    4233 s3context.go:194] found bucket in region "us-east-1"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x2309016]

goroutine 1 [running]:
k8s.io/kops/pkg/apis/kops/validation.validateEtcdClusterUpdate(0xc42095cba0, 0xc420f71780, 0xc420cc5100, 0x0, 0x5530e00, 0xc42095cb70, 0x0)
	/go/src/k8s.io/kops/pkg/apis/kops/validation/cluster.go:64 +0x66
k8s.io/kops/pkg/apis/kops/validation.ValidateClusterUpdate(0xc420963800, 0xc420cc5100, 0xc42046d800, 0x0, 0x0, 0x0)
	/go/src/k8s.io/kops/pkg/apis/kops/validation/cluster.go:47 +0x4e7
k8s.io/kops/pkg/client/simple/vfsclientset.(*ClusterVFS).Update(0xc421474910, 0xc420963800, 0xc420cc5100, 0xc4207a1000, 0x0, 0x0)
	/go/src/k8s.io/kops/pkg/client/simple/vfsclientset/cluster.go:139 +0xdf
k8s.io/kops/pkg/client/simple/vfsclientset.(*VFSClientset).UpdateCluster(0xc4208310a0, 0xc420963800, 0xc420cc5100, 0xc4207a1000, 0x0, 0x0)
	/go/src/k8s.io/kops/pkg/client/simple/vfsclientset/clientset.go:51 +0x4d
main.RunReplace(0xc4208309e0, 0xc420ad4280, 0x38b4be0, 0xc42000c018, 0xc420d8cab0, 0x0, 0x0)
	/go/src/k8s.io/kops/cmd/kops/replace.go:152 +0x518
main.NewCmdReplace.func1(0xc420ad4280, 0xc420d7cf40, 0x0, 0x4)
	/go/src/k8s.io/kops/cmd/kops/replace.go:80 +0x74
k8s.io/kops/vendor/github.com/spf13/cobra.(*Command).execute(0xc420ad4280, 0xc420d7ce80, 0x4, 0x4, 0xc420ad4280, 0xc420d7ce80)
	/go/src/k8s.io/kops/vendor/github.com/spf13/cobra/command.go:760 +0x2c1
k8s.io/kops/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x54f9b80, 0x2ba9400, 0x0, 0x0)
	/go/src/k8s.io/kops/vendor/github.com/spf13/cobra/command.go:846 +0x30a
k8s.io/kops/vendor/github.com/spf13/cobra.(*Command).Execute(0x54f9b80, 0x552fd90, 0x0)
	/go/src/k8s.io/kops/vendor/github.com/spf13/cobra/command.go:794 +0x2b
main.Execute()
	/go/src/k8s.io/kops/cmd/kops/root.go:97 +0x87
main.main()
	/go/src/k8s.io/kops/cmd/kops/main.go:25 +0x20
```

**6. What did you expect to happen?**

Replace the cluster template properly, and proceed to create a cluster

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

Again, I may be able to share privately, but I don't know what to greek out and what to not and not disrupt the failure.

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

```
kops replace --filename=/tmp/kluster-deploy/templates/kops/rendered/cluster.cspdev-red-us-west-2.cspdev.something.com.yaml --name=cspdev-red-us-west-2.cspdev.something.com --state=s3://st-cloud-cspdev --v=10
I1129 21:04:05.046279    4505 factory.go:68] state store s3://st-cloud-cspdev
I1129 21:04:05.055828    4505 aws_cloud.go:1094] Querying EC2 for all valid zones in region "us-west-2"
I1129 21:04:05.055998    4505 request_logger.go:45] AWS request: ec2/DescribeAvailabilityZones
I1129 21:04:05.561918    4505 status.go:57] Querying AWS for etcd volumes
I1129 21:04:05.561996    4505 status.go:68] Listing EC2 Volumes
I1129 21:04:05.563885    4505 request_logger.go:45] AWS request: ec2/DescribeVolumes
I1129 21:04:05.694633    4505 status.go:40] Cluster status (from cloud): {}
I1129 21:04:05.853344    4505 s3context.go:194] found bucket in region "us-east-1"
I1129 21:04:05.853400    4505 s3fs.go:220] Reading file "s3://pd-cloud-cspdev/cspdev-red-us-west-2.cspdev.something.com/config"
I1129 21:04:06.015316    4505 s3fs.go:220] Reading file "s3://pd-cloud-cspdev/cspdev-red-us-west-2.cspdev.something.com/config"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x2309016]

goroutine 1 [running]:
k8s.io/kops/pkg/apis/kops/validation.validateEtcdClusterUpdate(0xc420f2cc00, 0xc420fe0380, 0xc4207d32a0, 0x0, 0x5530e00, 0xc420f2cbd0, 0x0)
	/go/src/k8s.io/kops/pkg/apis/kops/validation/cluster.go:64 +0x66
k8s.io/kops/pkg/apis/kops/validation.ValidateClusterUpdate(0xc420f41400, 0xc4207d32a0, 0xc4207a6800, 0x0, 0x0, 0x0)
	/go/src/k8s.io/kops/pkg/apis/kops/validation/cluster.go:47 +0x4e7
k8s.io/kops/pkg/client/simple/vfsclientset.(*ClusterVFS).Update(0xc420d250e0, 0xc420f41400, 0xc4207d32a0, 0xc420079800, 0x0, 0x0)
	/go/src/k8s.io/kops/pkg/client/simple/vfsclientset/cluster.go:139 +0xdf
k8s.io/kops/pkg/client/simple/vfsclientset.(*VFSClientset).UpdateCluster(0xc4209ef060, 0xc420f41400, 0xc4207d32a0, 0xc420079800, 0x0, 0x0)
	/go/src/k8s.io/kops/pkg/client/simple/vfsclientset/clientset.go:51 +0x4d
main.RunReplace(0xc4209ee9a0, 0xc420e9e280, 0x38b4be0, 0xc4200cc008, 0xc420e80bd0, 0x0, 0x0)
	/go/src/k8s.io/kops/cmd/kops/replace.go:152 +0x518
main.NewCmdReplace.func1(0xc420e9e280, 0xc420eac080, 0x0, 0x4)
	/go/src/k8s.io/kops/cmd/kops/replace.go:80 +0x74
k8s.io/kops/vendor/github.com/spf13/cobra.(*Command).execute(0xc420e9e280, 0xc4209d9fc0, 0x4, 0x4, 0xc420e9e280, 0xc4209d9fc0)
	/go/src/k8s.io/kops/vendor/github.com/spf13/cobra/command.go:760 +0x2c1
k8s.io/kops/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x54f9b80, 0x2ba9400, 0x0, 0x0)
	/go/src/k8s.io/kops/vendor/github.com/spf13/cobra/command.go:846 +0x30a
k8s.io/kops/vendor/github.com/spf13/cobra.(*Command).Execute(0x54f9b80, 0x552fd90, 0x0)
	/go/src/k8s.io/kops/vendor/github.com/spf13/cobra/command.go:794 +0x2b
main.Execute()
	/go/src/k8s.io/kops/cmd/kops/root.go:97 +0x87
main.main()
	/go/src/k8s.io/kops/cmd/kops/main.go:25 +0x20
```

**9. Anything else do we need to know?**

--- a/pkg/apis/kops/validation/cluster.go
+++ b/pkg/apis/kops/validation/cluster.go
@@ -44,7 +44,9 @@ func ValidateClusterUpdate(obj *kops.Cluster, status *kops.ClusterStatus, old *k
 			fp := field.NewPath("spec", "etcdClusters").Key(k)
 
 			oldCluster := oldClusters[k]
-			allErrs = append(allErrs, validateEtcdClusterUpdate(fp, newCluster, status, oldCluster)...)
+			if oldCluster != nil {
+				allErrs = append(allErrs, validateEtcdClusterUpdate(fp, newCluster, status, oldCluster)...)
+			}
 		}
 		for k := range oldClusters {
 			newCluster := newClusters[k]

